### ISSUE TITLE
Building from coefficients

### ISSUE THREAD

**Issue by nlw0** [JuliaMath/Interpolations.jl#284]
Probably just an issue with my own understanding of things, but I would like to build a `BSplineInterpolation` object straight from coefficients, and I don't know what I'm missing. Is it even possible? What's the story with this first nameless parameter?

This is what I'm trying:

```
itp = interpolate(randn(5,5), BSpline(Cubic(Line(OnGrid()))))
itq = Interpolations.BSplineInterpolation(zeros(7,7), itp.parentaxes, itp.it)
```

Result:

> julia> itp = interpolate(randn(5,5), BSpline(Cubic(Line(OnGrid()))))
> 5×5 interpolate(OffsetArray(::Array{Float64,2}, 0:6, 0:6), BSpline(Cubic(Line(OnGrid())))) with element type Float64:
>   0.566049   1.95704   -2.06191     -1.08552    0.854445
>   0.243544   0.296544   0.00928176  -0.332793  -1.18356 
>   0.289149  -1.45364   -1.0586       1.81307    0.360731
>   0.858396  -0.58452   -0.0484449   -1.18305    1.4658  
>  -0.548378   0.898356   0.101132     0.226077   0.213026
> 
> julia> itq = Interpolations.BSplineInterpolation(zeros(7,7), itp.parentaxes, itp.it)
> ERROR: MethodError: no method matching Interpolations.BSplineInterpolation(::Array{Float64,2}, ::Tuple{Base.OneTo{Int64},Base.OneTo{Int64}}, ::BSpline{Cubic{Line{OnGrid}}})
> Closest candidates are:
>   Interpolations.BSplineInterpolation(::Type{TWeights<:Real}, ::AbstractArray{Tel,N}, ::IT<:Union{NoInterp, Tuple{Vararg{Union{NoInterp, BSpline},N} where N}, BSpline}, ::Any) where {N, Tel, TWeights<:Real, IT<:Union{NoInterp, Tuple{Vararg{Union{NoInterp, BSpline},N} where N}, BSpline}} at /home/user/.julia/packages/Interpolations/1CM1Q/src/b-splines/b-splines.jl:48
> Stacktrace:
>  [1] top-level scope at none:0
>

**Comment by timholy** [comment]
Looks like we're missing a constructor method that fills in all the type parameters. Care to submit a PR?

For the record the following works:

```julia
julia> BS = typeof(itp)
Interpolations.BSplineInterpolation{Float64,2,OffsetArrays.OffsetArray{Float64,2,Array{Float64,2}},BSpline{Cubic{Line{OnGrid}}},Tuple{Base.OneTo{Int64},Base.OneTo{Int64}}}

julia> BS(itp.coefs, itp.parentaxes, itp.it)
5×5 interpolate(OffsetArray(::Array{Float64,2}, 0:6, 0:6), BSpline(Cubic(Line(OnGrid())))) with element type Float64:
  2.08355    0.205414  -0.0308661   0.93061   -0.362805 
 -1.11214   -0.767313   0.134176   -0.230199   0.0718531
  1.87457   -0.114432  -1.67923    -0.317411  -0.89315  
 -2.23463    0.638659  -1.0083      1.60551    0.475472 
  0.584888  -0.677973  -0.280755    0.879349  -0.0452266
```

Be sure the `parentaxes` match the axes of the `coefs`; note that `itp.coefs` goes from 0:6 and `parentaxes` from 1:5, i.e., there is 1 padding element on each side.

**Comment by nlw0** [comment]
I made a PR #286

**Comment by dkarrasch** [comment]
This can be closed, it was fixed in #286.
